### PR TITLE
feat: mesh precompute functions and state generation

### PR DESCRIPTION
**Pull request by aranega**
This PR introduce all the function necessary to perform multi-scale mesh generation:

* from a precomputed segmentation project,
* for single mesh generation as precomputed format (standalone mesh without any segmentation data)
* for multiple mesh generation as precomputed format (standalone meshes for different labels, but without any segmenation data)

### CODE REVIEW

**Review by seankmartin — APPROVED** [review]
looks good!

File: cryoet_data_portal_neuroglancer/precompute/instance_mesh.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+LOGGER = logging.getLogger(__name__)
+
+
+def encode_oriented_mesh(
+    input_geometry: trimesh.Scene | trimesh.Trimesh,
+    data: list[dict[str, Any]],
+    max_lod: int = 2,
+    max_faces_for_first_lod: int = 5_000_000,
+    decimation_aggressiveness: float = 4.5,
```
Comment (manasaV3) [inline cryoet_data_portal_neuroglancer/precompute/instance_mesh.py:20]: nit: Is this supposed to be `5.0`?

**Review by manasaV3 — APPROVED** [review]

File: cryoet_data_portal_neuroglancer/precompute/instance_mesh.py (hunk 1 above)
Comment (seankmartin) [reply cryoet_data_portal_neuroglancer/precompute/instance_mesh.py:20]: Thanks Manasa, forgot to update the docstring! Will fix the docstring and leave the default here at 4.5

**Comment by seankmartin** [comment]
@manasaV3 should we merge this one with the 90 commits or would you prefer we squash into one commit and then merge? (cc @aranega)

**Pull request merged**